platform/orion/core/generic/committer/committer.go

```diff
@@ -312,13 +312,17 @@ func (c *committer) listenToFinality(txID string, timeout time.Duration) error {
 		iterations = 1
 	}
 	for i := 0; i < iterations; i++ {
+		timeout := time.NewTimer(c.pollingTimeout)
```

Comment:
wasn't this the case where After is okay?

Reply:
It's true that reading from the `ch` channel terminates the function, however I thought that this function can perhaps be called many times in a short period of time so it's better to stop the timer and not let them wait for being terminated with the timeout.